File: C++/src/Implementations/CellPhone.cpp
#include "CellPhone.h"
#include "GameState.h"
#include <iostream>
#include <cstdlib>

using std::cout;
using std::endl;
using std::to_string;

CellPhone::CellPhone() : bullet(bullet) {};

void CellPhone::UseItem(GameState& gameState) 
{
	bullet = rand() % gameState.magazine.GetMagazienSize();
	if ( gameState.magazine.CheckBulletTypeCellPhone(bullet) ) 
	{
		if (gameState.gameStateManager.GetStarter() == GameEnums::STARTER_HUMAN)
		{
			gameState.log.push_back("Uzyles telefonu (" + to_string(bullet + 1) + " pelna)!");
		}
		else
		{
			gameState.log.push_back("Komputer uzyl telefonu!");
		}
	}
	else
	{
		if (gameState.gameStateManager.GetStarter() == GameEnums::STARTER_HUMAN)
		{
			gameState.log.push_back("Uzyto telefonu (" + to_string(bullet + 1) + " pusta)!");
		}
		else
		{
			gameState.log.push_back("Komputer uzyl telefonu!");
		}
	}
}

File: C#/src/Implementations/CellPhone.cs
using System;

public class CellPhone : Item
{
    private int bullet = 0;
    private Random random = new Random();

    public CellPhone() { }

    public void UseItem(GameState gameState)
    {
        bullet = random.Next(gameState.magazine.GetMagazineSize());
        if (gameState.magazine.CheckBulletTypeCellPhone(bullet))
        {
            if (gameState.gameStateManager.GetStarter() == (int)GameEnums.Starter.STARTER_HUMAN)
            {
                gameState.log.Add("Uzyles telefonu (" + (bullet + 1).ToString() + " pelna)!");
            }
            else
            {
                gameState.log.Add("Komputer uzyl telefonu!");
            }
        }
        else
        {
            if (gameState.gameStateManager.GetStarter() == (int)GameEnums.Starter.STARTER_HUMAN)
            {
                gameState.log.Add("Uzyto telefonu (" + (bullet + 1).ToString() + " pusta)!");
            }
            else
            {
                gameState.log.Add("Komputer uzyl telefonu!");
            }
        }
    }
}